astropy/modeling/powerlaws.py

```diff
@@ -587,23 +587,26 @@ class Schechter1D(Fittable1DModel):
     alpha = Parameter(default=-1., description='Faint-end slope')
 
     @staticmethod
-    def evaluate(mag, phi_star, m_star, alpha):
+    def _factor(magnitude, m_star):
+        factor_exp = (m_star - magnitude)
+        if isinstance(factor_exp, Quantity):
+            return (-1 * factor_exp).to(dimensionless_unscaled)
+        else:
+            return 10 ** (0.4 * factor_exp)
```

Comment:
```suggestion
        factor_exp = (magnitude - m_star)
        if isinstance(factor_exp, Quantity):
            return factor_exp.to(dimensionless_unscaled)
        else:
            return 10 ** (-0.4 * factor_exp)
```

This seems easier to read/understand to me.